kivymd/uix/button/button.py

```diff
@@ -903,333 +979,96 @@ def stop_elevation_anim(self):
-
-    :attr:`line_width` is an :class:`~kivy.properties.NumericProperty`
-    and defaults to `1`.
+class MDFlatButton(ButtonContentsText, BaseButton):
+    """A flat rectangular button with (by default) no border or
+    background. Text is the default text color.
     """
+    pass
 
-    line_color = ColorProperty(None)
-    """
-    Line color for button border.
 
-    :attr:`line_color` is an :class:`~kivy.properties.ColorProperty`
-    and defaults to `None`.
+class MDRaisedButton(
+    FakeRectangularElevationBehavior, ButtonElevationBehaviour,
+    ButtonContentsText, BaseButton
+):
+    """A flat button with (by default) a primary color fill and matching
+    color text.
     """
-
-    md_bg_color = ColorProperty([0.0, 0.0, 0.0, 0.0])
-
-    def __init__(self, **kwargs):
-        super().__init__(**kwargs)
-        Clock.schedule_once(self.set_text_color)
-
-
-class MDRaisedButton(BaseRectangularButton, BaseElevationButton):
-    theme_text_color = StringProperty("Custom")
-
-    def __init__(self, **kwargs):
-        super().__init__(**kwargs)
-        Clock.schedule_once(self.update_text_color)
-
-    def update_text_color(self, *args) -> NoReturn:
-        if self.text_color in (
-            [0.0, 0.0, 0.0, 0.87],
-            [0.0, 0.0, 0.0, 1.0],
-            [1.0, 1.0, 1.0, 1.0],
-        ):
-            self.text_color = text_colors[self.theme_cls.primary_palette][
-                self.theme_cls.primary_hue
-            ]
-
-
-class MDFlatButton(BaseFlatButton):
-    md_bg_color = ColorProperty([0.0, 0.0, 0.0, 0.0])
-
-
-class MDRectangleFlatButton(BaseRectangleFlatButton):
-    # TODO: If the user has set a custom text color, then when the application
-    #  color palette changes, the text color will also change to the current
-    #  color of the color scheme. Do need to preserve custom text colors when
-    #  changing the application palette?
-
-    def __init__(self, **kwargs):
-        super().__init__(**kwargs)
-        Clock.schedule_once(self._remove_shadow)
+    _default_md_bg_color = None
+    _default_md_bg_color_disabled = None
+    _default_theme_text_color = "Custom"
+    _default_text_color = "PrimaryHue"
 
 
-class MDRectangleFlatIconButton(BaseRectangleFlatButton):
-    icon = StringProperty("android")
+class MDRectangleFlatButton(ButtonContentsText, BaseButton):
+    """A flat button with (by default) a primary color border and primary
+    color text.
     """
-    Button icon.
+    _default_line_color = None
+    _default_line_color_disabled = None
+    _default_theme_text_color = "Custom"
+    _default_text_color = "Primary"
 
-    :attr:`icon` is an :class:`~kivy.properties.StringProperty`
-    and defaults to `'android'`.
-    """
 
-    icon_color = ColorProperty(None)
+class MDRectangleFlatIconButton(ButtonContentsIconText, BaseButton):
+    """A flat button with (by default) a primary color border,
+    primary color text and a primary color icon on the left.
     """
```

Comment:
"""
    A flat button with (by default) a primary color border,
    primary color text and a primary color icon on the left.
    """